models/transaction.go

```diff
@@ -0,0 +1,106 @@
+package models
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/onflow/cadence"
+	cdcCommon "github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/flow-go/fvm/evm/types"
+	"math/big"
+)
+
+var txExecutedType = (types.EVMLocation{}).TypeID(nil, string(types.EventTypeTransactionExecuted))
+
+type txEventPayload struct {
+	BlockHeight             uint64 `cadence:"blockHeight"`
+	TransactionHash         string `cadence:"transactionHash"`
+	Transaction             string `cadence:"transaction"`
+	TransactionType         int    `cadence:"transactionType"`
+	Failed                  bool   `cadence:"failed"`
+	GasConsumed             uint64 `cadence:"gasConsumed"`
+	DeployedContractAddress string `cadence:"deployedContractAddress"`
+	ReturnedValue           string `cadence:"returnedValue"`
+	Logs                    string `cadence:"logs"`
+}
+
+// DecodeReceipt takes a cadence event for transaction executed and decodes it into the receipt.
+func DecodeReceipt(event cadence.Event) (*gethTypes.Receipt, error) {
+	if cdcCommon.TypeID(event.EventType.ID()) != txExecutedType {
+		return nil, fmt.Errorf(
+			"invalid event type for decoding into receipt, received %s expected %s",
+			event.Type().ID(),
+			types.EventTypeTransactionExecuted,
+		)
+	}
+
+	var tx txEventPayload
+	err := cadence.DecodeFields(event, &tx)
+	if err != nil {
+		return nil, err
+	}
+
+	encLogs, err := hex.DecodeString(tx.Logs)
+	if err != nil {
+		return nil, err
+	}
+
+	var logs []*gethTypes.Log
+	err = rlp.Decode(bytes.NewReader(encLogs), &logs)
+	if err != nil {
+		return nil, err
+	}
+
+	receipt := &gethTypes.Receipt{
+		Type:              0,              // todo check
```

Comment:
were you planning to address these `todo`s in this PR or a future one? is there an issue tracking them?

Reply:
https://github.com/onflow/flow-evm-gateway/issues/87